API_DOCUMENTATION.md

```diff
@@ -0,0 +1,414 @@
+
+```
+npm install nucleojs --save
+```
+
+Using Yarn:
+
+```
+yarn add nucleojs
+```
+
+## Usage
+
+```javascript
+import {
+  NucleoString,
+  NucleoNumber,
+  NucleoBoolean,
+  NucleoObject,
+  createStore
+} from 'nucleojs';
+
+const fullNameContract = new NucleoObject({
+  name: 'fullName',
+  fields:  {
+    firstName: NucleoString,
+    lastName: NucleoString
+  }
+});
+
+const userContract = new NucleoObject({
+  name: 'user',
+  fields: {
+    name: fullNameContract,
+    age: NucleoNumber,
+    disabled: NucleoBoolean
+  }
+});
+
+const productsContract = new NucleoObject({
+  name: 'products',
+  fields: {
+    title: NucleoString,
+    price: NucleoNumber
+  }
+});
+
+const contracts = {
+  user: userContract,
+  products: productsContract
+};
+
+const store = createStore(contracts);
+const { dispatch, update, getStore, subscribe } = store;
+
+const user = dispatch('user')({ name: { firstName: 'John', lastName: 'Nor' }, age: 27 });
+// it'll save the data to store properly
+
+console.log(user);
+/*
+{
+  status: 'OK',
+  errors: [],
+  data: {
+    name: {
+      firstName: 'John',
+      lastName: 'Nor'
+    },
+    age: 27
+  },
+}
+*/
+```
+
+## Contracts
+
+### Todo
+
+## Types
+
+### Creating NucleoObject
+
+```javascript
+NucleoObject({
+  name: String,
+  fields:  NucleoObject
+});
+```
+
+**Usage**
+
+```javascript
+import {
+  NucleoString,
+  NucleoNumber,
+  NucleoBoolean,
+  NucleoObject
+} from 'nucleojs'
+
+const fullNameContract = new NucleoObject({
+  name: 'fullName',
+  fields:  {
+    firstName: NucleoString,
+    lastName: NucleoString
+  }
+});
+
+const userContract = new NucleoObject({
+  name: 'user', // don't need to be the same name as the variable, but need to be unique
+  fields: {
+    name: fullNameContract,
+    age: NucleoNumber,
+    disabled: NucleoBoolean
+  }
+});
+
+const productsContract = new NucleoObject({
+  name: 'products',
+  fields: {
+    title: NucleoString,
+    price: NucleoNumber
+  }
+});
+
+const contracts = {
+  user: userContract,
+  products: productsContract
+};
+```
+
+### Creating NucleoList
+
+```javascript
+NucleoObject(NucleoObject);
+```
+
+**Usage**
+
+```javascript
+import {
+  NucleoString,
+  NucleoNumber,
+  NucleoBoolean,
+  NucleoObject,
+  NucleoList
+} from 'nucleojs'
+
+const fullNameContract = new NucleoObject({
+  name: 'fullName',
+  fields:  {
+    firstName: NucleoString,
+    lastName: NucleoString
+  }
+});
+
+const userContract = new NucleoObject({
+  name: 'user', // don't need to be the same name as the variable, but need to be unique
+  fields: {
+    name: fullNameContract,
+    age: NucleoNumber,
+    disabled: NucleoBoolean,
+  }
+});
+
+const usersContract = new NucleoList(userContract);
+
+const itemsContract = new NucleoList(NucleoString);
+
+const contracts = {
+  users: usersContract,
+  items: itemsContract
+};
+```
+
+## Creating NucleoString
+
+```javascript
+NucleoString: String;
+```
+
+**Usage**
+
+```javascript
+const name = 'Jhon';
```

Comment:
We just can use it inside a `NucleoObject`, what about we put the whole model of usage here like:

```suggestion
const personalInfoType = new NucleoObject({
  name: 'personalInfo',
  fields:  {
    firstName: NucleoString,
    lastName: NucleoString
  }
});
```